pkg/credentialprovider/plugin/config.go

```diff
@@ -126,3 +137,54 @@ func validateCredentialProviderConfig(config *kubeletconfig.CredentialProviderCo
 
 	return allErrs
 }
+
+// appendSystemEnvVars appends provided array of strings of environment variables in the form of KEY=VALUE
+// with the env vars present in the config
+func appendSystemEnvVars(systemEnvVars []string, config *config.CredentialProvider) {
+	if config == nil {
+		return
+	}
+	configEnvVarNames := getEnvVarKeys(config.Env)
```

Comment:
Can config.Env ever be nil? Should that be checked?

Reply:
Hello @MorrisLaw , Thanks for taking out time to review this PR. 

Yes `config.Env` can be `nil` however `getEnvVarKeys` would return an empty map in that case and I verified that the final result would be that `CredentialProviderConfig` would only have system env vars.

Apparently, calling range on `nil` map results in an empty iteration - https://play.golang.org/p/jvhzUKKwMCO

Though I am prepared to check for `nil` in order to prevent code smell

Reply:
Thanks for looking into that. It’s always good to confirm 😄 But yeah your approach seems reasonable 👍🏽